### PR TITLE
Lex fodder in a manner compatible with C++

### PR DESCRIPTION
**Pull request by sparkprime**
I also rewrote lexer_test to use individual tests for each case instead of iterating through an array.  While not as dense on the page, the density of information is about the same and this way you can get more useful stacktraces as well as easily select specific tests to run.

### CODE REVIEW

**Comment by sparkprime** [comment]
The tests didn't build because of a stray file (not in the PR) and looked like they succeeded but actually there is quite a lot more work for me to do here

**Comment by sparkprime** [comment]
Ok it actually wasn't that hard.

File: ast/fodder.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	// interstitial, it will also flow after it on the same line.  If it follows
+	// a new line or a paragraph, it is the first thing on the following line,
+	// after the blank lines and indentation specified by the previous fodder.
+	//
+	// There is exactly one comment string.
+	FodderInterstitial
+
+	// FodderParagraph represents a comment consisting of at least one line.
+	//
+	// // and # style commes have exactly one line.  C-style comments can have
```
Comment (sbarzowski) [inline ast/fodder.go]: commes -> comments

File: parser/lexer.go [hunk 2]
```diff
@@ -209,6 +188,47 @@ func isSymbol(r rune) bool {
 	return false
 }
 
+func isHorzWs(r rune) bool {
```
Comment (sbarzowski) [inline parser/lexer.go]: nit: I wouldn't abbreviate it. I would go with`isHorizontalWhitespace`.

Comment (sbarzowski) [reply parser/lexer.go]: Same with other functions with "Ws" in their names.

File: parser/lexer_test.go [hunk 3]
```diff
@@ -272,24 +193,628 @@ func tokensEqual(ts1, ts2 Tokens) bool {
...
+			errString, expectedError)
+	}
+	if err == nil && !tokensEqual(tokens, testTokens) {
+		t.Errorf("got\n\t%+v\nexpected\n\t%+v", tokens, expected)
+	}
+}
+
+// TODO: test position reporting
+
+func TestEmpty(t *testing.T) {
```
Comment (sbarzowski) [inline parser/lexer_test.go]: What was wrong with the old table-style tests?

Comment (sparkprime) [reply parser/lexer_test.go]: Impossible to run a single test in isolation without modifying the code.  Although Go doesn't seem to do this stuff yet, other frameworks prefer breaking stuff up as much as possible to enable individual running of tests, reporting of errors at finer granularity, running tests in parallel, and helping avoid state accumulating between tests.

File: ast/fodder.go (hunk 1 above)
Comment (sparkprime) [reply ast/fodder.go]: Done.

File: parser/lexer.go (hunk 2 above)
Comment (sparkprime) [reply parser/lexer.go]: Done.

File: parser/lexer_test.go (hunk 3 above)
Comment (sbarzowski) [reply parser/lexer_test.go]: I'm ok with that.

Just FYI the way it worked before was **not** the best you could get from Table Driven Tests. All of the stuff you mentioned can be achieved with TDT. Most importantly it lacked `t.Run(tt.in, func(t *testing.T) {...` - so all the cases were treated as one test. If we added that one line with `t.Run(...` each testcase would be a subtest and we could easily run them separately and in parallel.
More info: https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks

**Pull request merged**